features/steps/helper.go

```diff
@@ -0,0 +1,316 @@
+func (f *JobsFeature) checkTaskStructure(id string, lastUpdated time.Time, expectedResult map[string]string, links *models.TaskLinks, taskName string) error {
+	_, err := uuid.FromString(id)
+	if err != nil {
+		return fmt.Errorf("the jobID should be a uuid: %w", err)
+	}
+
+	if lastUpdated.After(time.Now()) {
+		return errors.New("expected LastUpdated to be now or earlier but it was: " + lastUpdated.String())
+	}
+
+	expectedLinksJob := strings.Replace(expectedResult["links: job"], "{bind_address}", f.Config.BindAddr, 1)
+	expectedLinksJob = strings.Replace(expectedLinksJob, "{id}", id, 1)
+
+	assert.Equal(&f.ErrorFeature, expectedLinksJob, links.Job)
+
+	expectedLinksSelf := strings.Replace(expectedResult["links: self"], "{bind_address}", f.Config.BindAddr, 1)
+	expectedLinksSelf = strings.Replace(expectedLinksSelf, "{id}", id, 1)
+	expectedLinksSelf = strings.Replace(expectedLinksSelf, "{task_name}", taskName, 1)
+
+	assert.Equal(&f.ErrorFeature, expectedLinksSelf, links.Self)
+	return nil
+}
+
+// checkValuesInJob is a utility function that can be called by a feature step in order to check that the values
+// of certain attributes, in a job, are all equal to the expected ones.
+func (f *JobsFeature) checkValuesInJob(expectedResult map[string]string, job models.Job) {
+	assert.Equal(&f.ErrorFeature, expectedResult["number_of_tasks"], strconv.Itoa(job.NumberOfTasks))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_completed"], job.ReindexCompleted.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_failed"], job.ReindexFailed.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["reindex_started"], job.ReindexStarted.Format(time.RFC3339))
+	assert.Equal(&f.ErrorFeature, expectedResult["state"], job.State)
+	assert.Equal(&f.ErrorFeature, expectedResult["total_search_documents"], strconv.Itoa(job.TotalSearchDocuments))
+	assert.Equal(&f.ErrorFeature, expectedResult["total_inserted_search_documents"], strconv.Itoa(job.TotalInsertedSearchDocuments))
+}
+
+// checkValuesInTask is a utility function that can be called by a feature step in order to check that the values
+// of certain attributes, in a task, are all equal to the expected ones.
+func (f *JobsFeature) checkValuesInTask(expectedResult map[string]string, task models.Task) {
+	assert.Equal(&f.ErrorFeature, expectedResult["number_of_documents"], strconv.Itoa(task.NumberOfDocuments))
+	assert.Equal(&f.ErrorFeature, expectedResult["task_name"], task.TaskName)
+}
+
+// readOutputMessages is a utility method to read the kafka messages that get sent to the producer's output channel
+func (f *JobsFeature) readOutputMessages() {
+	go func() {
+		for {
+			select {
+			case f.kafkaProducerOutputData <- <-f.MessageProducer.Channels().Output:
+				log.Println("read")
+			case <-f.quitReadingOutput:
+				return
+			}
+		}
+	}()
+}
+
+func readAndDeserializeKafkaProducerOutput(kafkaProducerOutputData <-chan []byte) (*models.ReindexRequested, error) {
```

Comment:
Im not sure i understand why this is written like this, we have a for loop but only ever read off the first item before returning, is there any point in the for loop?

Is it better to check length and if empty return nil otherwise get the first item off the array?

Reply:
I did not implement this function but you are completely right. I don't understand why it is done this way. 
The function isn't even getting the first item off the array. It's just getting the `[]byte` from the channel and then unmarshalling that. 

I have updated the function anyway and the component testing is passing